src/inmanta/server/agentmanager.py

```diff
@@ -264,6 +276,32 @@ async def _unpause_agent(
                 env.id, endpoints_with_new_primary, now=datetime.now().astimezone(), connection=connection
             )
 
+    async def _keep_agent_paused_on_resume(
+        self, env: data.Environment, endpoint: Optional[str] = None, connection: Optional[asyncpg.connection.Connection] = None
+    ) -> None:
+        """
+        Set the unpause_on_resume field of an agent (or all agents in an environment when the endpoint is set to None)
+        so that the agent is still paused after the environment is resumed
+        """
+
+        async with self.session_lock:
+            await data.Agent.set_unpause_on_resume(
+                env=env.id, endpoint=endpoint, should_be_unpaused_on_resume=False, connection=connection
+            )
+
+    async def _unpause_agent_on_resume(
+        self, env: data.Environment, endpoint: Optional[str] = None, connection: Optional[asyncpg.connection.Connection] = None
+    ) -> None:
+        """
+        Set the unpause_on_resume field of an agent (or all agents in an environment when the endpoint is set to None)
+        so that the agent is unpaused after the environment is resumed
+        """
+
+        async with self.session_lock:
+            await data.Agent.set_unpause_on_resume(
+                env=env.id, endpoint=endpoint, should_be_unpaused_on_resume=True, connection=connection
+            )
+
```

Comment:
This is two times the same method. Only the boolean flag differs. Why not make one method and parameterize it with an argument?